spki/src/algorithm.rs

```diff
@@ -122,7 +122,7 @@ impl<'a> AlgorithmIdentifierRef<'a> {
         if actual_oid == expected_oid {
             Ok(actual_oid)
         } else {
-            Err(Error::OidUnknown { oid: expected_oid })
+            Err(Error::OidUnknown { oid: actual_oid })
```

Comment:
These could probably use tests.

Perhaps the it should also carry the expected OID if available.

Reply:
adding the expected should be cheap.